### PR TITLE
tests(smoke): check for passing robots-txt

### PR DESCRIPTION
**Pull request by connorjclark**
would have prevented #12936

### CODE REVIEW

File: lighthouse-core/gather/gatherers/seo/robots-txt.js [hunk 1]
```diff
@@ -8,13 +8,13 @@
...
 
-/* global fetch, location */
+/* global fetch, location, window */
 
 /** @return {Promise<LH.Artifacts['RobotsTxt']>} */
 /* c8 ignore start */
 async function getRobotsTxtContent() {
   try {
-    const response = await fetch(new URL('/robots.txt', location.href).href);
+    const response = await fetch(new window.__nativeURL('/robots.txt', location.href).href);
```
Comment (connorjclark) [inline lighthouse-core/gather/gatherers/seo/robots-txt.js:17]: took awhile to figure out why the smoke test wasn't passing, it's because seo tester html overrides `URL`! so, luckily we get a driveby fix here for naughty pages overriding globals.

Comment (patrickhulce) [reply lighthouse-core/gather/gatherers/seo/robots-txt.js:17]: hm, `_cachedNativesPreamble` should be solving this, is something broken there? that's more concerning than the smoke itself :/

https://github.com/GoogleChrome/lighthouse/blob/ce870607f8bbcabdaf3292b30f759e69f36cb9c3/lighthouse-core/gather/driver/execution-context.js#L226-L235

Comment (connorjclark) [reply lighthouse-core/gather/gatherers/seo/robots-txt.js]: ok so this seems fine and i can't get it to fail again, weird

**Review by patrickhulce — APPROVED** [review]

**Comment by connorjclark** [comment]
@patrickhulce / @adamraine  any idea why FR is failing?

**Comment by adamraine** [comment]
Could it be flaky? I'm not seeing a failure locally on the smoke test or a direct FR navigation of the test page.
